pkg/controller/defaults/testdata/volumeclass/volume-class-defaults-same-gen/expected.golden

```diff
@@ -30,13 +30,14 @@ status:
   appStatus: {}
   columns: {}
   conditions:
+  - error: true
+    message: 'cannot establish defaults because two defaults volume classes exist:
+      test-volume-class and test-volume-class-1'
```

Comment:
Is this the desired outcome for this test?

Reply:
In this case, yes. We are not longer only processing the defaults when the generation is different. Instead we are processing it always if a default has not been set for this Volume. As such this test is working as expected given that the input had two default VolumeClasses set this causing a conflict. That might be too confusing of a test though and this could instead be replaced with something more inline with what we expect to happen

Reply:
Is that the desired behavior? My understanding was we don't want to potentially break apps unless a user took an explicit action.

In other words, I understand that we only want to set the volume default once, but should that only happen once an existing app is updated?

Reply:
Talked offline but to recap - this is a temporary thing that is necessary while we're migrating away from VolumeSize. The logic is essentially: if VolumeSize is set, make sure it is set on the Volumes map. This needs to happen at least once and with it being behind the generation would mean a user would have to explicitly update their app to get that Default set on the Volumes map. Once this merges and propagates we will be able to put it back behind the generation like we'd expect.